Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -612,7 +645,57 @@ void ReflectometryReductionOne::exec() {
   }
 
   IvsQ = this->toIvsQ(IvsLam, correctDetectorPositions, theta, isPointDetector);
-
+  double momentumTransferMinimum = getProperty("MomentumTransferMinimum");
+  double momentumTransferStep = getProperty("MomentumTransferStep");
+  double momentumTransferMaximum = getProperty("MomentumTransferMaximum");
+  MantidVec QParams;
+  if (isDefault("MomentumTransferMinimum"))
+    momentumTransferMinimum = calculateQ(IvsLam->readX(0).back(), theta.get());
+  if (isDefault("MomentumTransferMaximum"))
+    momentumTransferMaximum = calculateQ(IvsLam->readX(0).front(), theta.get());
+  if (isDefault("MomentumTransferStep")) {
+    // if the DQQ is not given for this run.
+    // we will use CalculateResoltion to produce this value
+    // for us.
+    IAlgorithm_sptr calcResAlg =
+        AlgorithmManager::Instance().create("CalculateResolution");
+    calcResAlg->setProperty("Workspace", runWS);
+    calcResAlg->setProperty("TwoTheta", theta.get());
```

Comment:
Again the confusion between `TwoTheta` and `theta`. Here we are setting the property `TwoTheta` to our angle. This seems to be wrong, but I have checked that this is what we are doing in the interface (see [here](https://github.com/mantidproject/mantid/blob/master/MantidQt/CustomInterfaces/src/Reflectometry/ReflTableViewPresenter.cpp#L549-L557)).